src/transactionsFactories/tokenManagementTransactionIntentsFactory.spec.ts

```diff
@@ -218,4 +219,158 @@ describe("test token management transactions factory", () => {
         assert.equal(transaction.receiver, grace.address.toString());
         assert.equal(transaction.value, 0n);
     });
+
+    it("should create 'Transaction' for modifying royalties", () => {
+        const transaction = tokenManagementFactory.createTransactionForModifyingRoyalties({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            royalties: 1234n,
+        });
+
+        assert.deepEqual(transaction.data, Buffer.from("ESDTModifyRoyalties@544553542d313233343536@01@04d2"));
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60125000n);
+    });
+
+    it("should create 'Transaction' for setting new uris", () => {
+        const transaction = tokenManagementFactory.createTransactionForSettingNewUris({
+            sender: grace.address,
+            tokenIdentifier: "TEST-123456",
+            tokenNonce: 1n,
+            uris: ["firstURI", "secondURI"],
+        });
+
+        assert.deepEqual(
+            transaction.data,
+            Buffer.from("ESDTSetNewURIs@544553542d313233343536@01@6669727374555249@7365636f6e64555249"),
+        );
+        assert.equal(transaction.sender, grace.address.toString());
+        assert.equal(transaction.receiver, grace.address.toString());
+        assert.equal(transaction.value, 0n);
+        assert.equal(transaction.gasLimit, 60164000n);
+    });
+
+    it("should create 'Transaction' for modifying creator", () => {
+        const transaction = tokenManagementFactory.createTransactionForModifyingCreator({
+            sender: grace.address,
```

Comment:
Just to double check, only the current creator can do this? If so, who would be the new creator?

Reply:
from the PR on docs:
```
The creator of a token can be changed. For this, the token has to be moved to the new creator account. The new creator
account requires `ESDTRoleModifyCreator` role. Also, the token has to be of dynamic type.
```